File: baekjoon/bronze/level1/test_1157/java/Main.java
package baekjoon.bronze.level1.test_1157.java;

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String input = br.readLine().toUpperCase();
        Map<String, Integer> map = new HashMap<>();

        for (int i = 0; i < input.length(); i++) {
            if (!map.containsKey(Character.toString(input.charAt(i)))) {
                map.put(String.valueOf(input.charAt(i)), input.length() - input.replace(String.valueOf(input.charAt(i)), "").length());
            }
        }

        List<Map.Entry<String, Integer>> list = new LinkedList<>(map.entrySet());
        list.sort(Map.Entry.comparingByValue());

        if (list.size() > 1 && Objects.equals(list.get(list.size() - 1).getValue(), list.get(list.size() - 2).getValue()))
            bw.write("?");
        else bw.write(list.get(list.size() - 1).getKey());
        bw.close();
    }

}

File: baekjoon/bronze/level1/test_1157/kotlin/Main.kt
package bronze.level1.test_1157.kotlin

fun main() {
    val input = readln().toCharArray()
    val map = mutableMapOf<Char, Int>()

    var maxKey = ' '
    var maxCount = 0
    for (c in input) {
        val key = c.uppercaseChar()
        map[key] = map.getOrDefault(key, 0) + 1

        if (maxCount < map[key]!!) {
            maxKey = key
            maxCount = map[key]!!
        }
    }

    for (key in map.keys) {
        if (map[key]!! >= maxCount && key != maxKey) {
            maxKey = '?'
        }
    }

    print(maxKey)
}
